docs/docs/guides/multi-stage-delivery.md

```diff
@@ -0,0 +1,258 @@
+
+```shell
+git add .
+git commit -m "add application manifests"
+git push
+```
+
+After adding the files to the repository,
+it is time to create the Argo CD applications,
+one for the `dev` stage, and one for the `production`
+stage.
+These applications will point to the subdirectories
+containing the helm charts for the stages within the upstream
+git repository.
+To create the applications on Argo CD, apply the following manifests:
+
+```yaml title="apps.yaml"
+{% include "./assets/multi-stage-delivery/argo-apps.yaml" %}
+```
+
+Once the manifest above has been applied to the cluster,
+Argo CD will eventually synchronize with the upstream repository
+and apply the application helm charts located in the upstream repository.
+Note that initially, both the `dev` and `production` stages will be deployed
+after creating the Argo CD applications.
+Subsequent deployments of a new application version however will
+always be deployed by first updating them in the `dev` stage first, and
+then only by merging the PRs created by the promotion task will
+they be deployed into `production`.
+
+After this initial deployment, there will be a `KeptnAppVersion`
+in the `dev` stage:
+
+```shell
+$ kubectl get keptnappversion -n podtato-head-dev
+
+NAME                           APPNAME        VERSION   PHASE
+podtato-head-v0.3.0-6b86b273   podtato-head   v0.3.0    Completed
+```
+
+And there will be another `KeptnAppVersion` in the `production` stage.
+
+```shell
+$ kubectl get keptnappversion -n podtato-head-production
+
+NAME                           APPNAME        VERSION   PHASE
+podtato-head-v0.3.0-6b86b273   podtato-head   v0.3.0    Completed
+```
+
+## Promoting a workload
+
+Now that everything is set up, it is time to deploy a new version into `dev`
+and automatically promote it into `production` if the deployment into `dev`
+has been successful.
+To do so, change the `serviceVersion` value in `dev/values.yaml` from `v0.3.0`
+to `v0.3.1`:
+
+```yaml title="dev/values.yaml"
+{% include "./assets/multi-stage-delivery/values-dev-updated.yaml" %}
+```
+
+After changing the value, commit and push the updated file to
+the upstream repository:
+
+```shell
+git add dev/values.yaml
+git commit -m "deploy new version into dev"
+git push
+```
+
+Eventually, Argo CD will notify these changes and apply the updated helm chart
+to the cluster.
+This is also reflected in a new `KeptnAppVersion` being created in the `dev` stage:
+
+```shell
+$ kubectl get keptnappversion -n podtato-head-dev
+
+NAME                           APPNAME        VERSION   PHASE
+podtato-head-v0.3.0-6b86b273   podtato-head   v0.3.0    Completed
+podtato-head-v0.3.1-d4735e3a   podtato-head   v0.3.1    Completed
+```
+
+After the new version has been deployed, and the post deployment phase
+is completed, the `KeptnTask` executed in the promotion phase
+triggers a GitHub action that creates a pull request for applying the
+updated values to the helm chart of the `production` stage.
```

Comment:
I would put this into a numbered list with the steps that happen. I think that will be easier to read.